osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -218,7 +216,8 @@ public Task ChangeSettings(Optional<string> name = default, Optional<PlaylistIte
                 BeatmapChecksum = item.GetOr(existingPlaylistItem).Beatmap.Value.MD5Hash,
                 RulesetID = item.GetOr(existingPlaylistItem).RulesetID,
                 RequiredMods = item.HasValue ? item.Value.AsNonNull().RequiredMods.Select(m => new APIMod(m)).ToList() : Room.Settings.RequiredMods,
-                AllowedMods = item.HasValue ? item.Value.AsNonNull().AllowedMods.Select(m => new APIMod(m)).ToList() : Room.Settings.AllowedMods
+                AllowedMods = item.HasValue ? item.Value.AsNonNull().AllowedMods.Select(m => new APIMod(m)).ToList() : Room.Settings.AllowedMods,
+                PlaylistItemId = Room.Settings.PlaylistItemId,
```

Comment:
I'm a bit perplexed with this set. This is a server call to change current room settings, right? If so, then if I'm thinking correctly, it's in the server's purview (and best interest) to ignore this incoming value and create an entirely new playlist item. Does it make sense to even send this over the wire, then? It's slightly confusing when reading.

Not suggesting to split separate models for set and get, would just remove this assignment and send the default value across.

Reply:
As I mentioned in the osu-server-spectator PR, I don't necessarily see this (having PlaylistItemId mutable) as a bad thing for tournament mode where it's up to the clients to select.

But I do agree with everything else. This set, especially as it's coming from `Room.Settings.PlaylistItemId` is really weird. So...
1. For normal mode, the server ignores this value and takes on the incoming beatmap/ruleset/mods.
2. For tournament mode, the server takes on this value and validates beatmap/ruleset/mods.
3. For "host rotate" mode, the server takes on none of these values.

Does that sound fine? Honestly it's probably better as a different structure altogether that is separate from e.g. changing the room name, but I'm not sure about going down that path yet?

Reply:
Right, that outline of things to come makes things clearer. Sure, I think the suggested variants sound fine.

Would also lean towards agreeing with separate structures, but that's out of scope of this particular pull.